expan/core/experiment.py

```diff
@@ -236,6 +236,21 @@ def sga(self, feature_name_to_bins, multi_test_correction=False):
 
         return subgroups
 
+    def _isNotValidForAnalysis(self, df):
```

Comment:
just a minor thing: why not `_isValidForAnalysis`?

Reply:
```_isValidForAnalysis``` is a bit better. True.

Well, those general checks will still be run when we create an ```Experiment``` object regardlessly, so they are already verified. This method in this PR is an additional check when we are in ```sga```.

Reply:
What I was saying is that no-one prevents you from calling `_isNotValidForAnalysis` with a totally random, `Experiment`-instance unrelated dataframe

Reply:
But I guess it's a consequence of our design decision to split dataframes for SGA. Alternatively, we could span daughter instances of `Experiment` for each SGA sub-test.

Reply:
Got it. I recall the discussion now. We should keep track of technical debts.